### PR TITLE
rename delegate from $next (or $frame) to $delegate

### PR DESCRIPTION
**Pull request by mindplay-dk**
per #15

### CODE REVIEW

File: src/MiddlewareInterface.php [hunk 1]
```diff
@@ -8,15 +8,13 @@
 interface MiddlewareInterface
 {
     /**
-     * Process a client request and return a response.
-     *
-     * Takes the incoming request and optionally modifies it before delegating
-     * to the next frame to get a response.
+     * Process an incoming client or server request and return a response,
```
Comment (shadowhand) [inline src/MiddlewareInterface.php:11]: This first line should be a short summary, followed by a longer description. As such, I think the original "title line" was correct, and this change should replace the description.

File: src/ServerMiddlewareInterface.php [hunk 2]
```diff
@@ -8,15 +8,13 @@
 interface ServerMiddlewareInterface
 {
     /**
-     * Process a server request and return a response.
-     *
-     * Takes the incoming request and optionally modifies it before delegating
-     * to the next frame to get a response.
+     * Process an incoming server request and return a response, optionally delegating
+     * to the next middleware component to create the response.
```
Comment (shadowhand) [inline src/ServerMiddlewareInterface.php:12]: Similarly here.

File: src/MiddlewareInterface.php [hunk 3]
```diff
@@ -8,15 +8,13 @@
...
+     * optionally delegating to the next middleware component to create the response.
      *
      * @param RequestInterface $request
-     * @param DelegateInterface $next
+     * @param DelegateInterface $delegate
```
Comment (shadowhand) [inline src/MiddlewareInterface.php:15]: 👍

**Comment by shadowhand** [comment]
@mindplay-dk are you going to make the changes I suggested?

**Comment by mindplay-dk** [comment]
It's a two-line summary.

According to the [PSR-5 draft](https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#51-summary):

> A Summary MUST contain an abstract of the "Structural Element" defining the purpose. It is RECOMMENDED for Summaries to span a single line or at most two but not more than that.

It's a valid doc-block.

Rephrasing a slightly longer repetition of the same information just creates ambiguity.

**Comment by shadowhand** [comment]
:+1: 

[Approved with PullApprove](https://pullapprove.com/http-interop/http-middleware/pull-request/18/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=http-interop/http-middleware)

**Pull request merged**